### PR TITLE
Add openshift-apiserver as dependency for OCM

### LINKED ISSUE #515: [BUG] openshift-controller-manager should depend on openshift-apiserver
#### What happened:
`openshift-controller-manager` starts directly after `kube-apiserver` is ready without waiting for `openshift-apiserver` to be ready, too:
https://github.com/redhat-et/microshift/blob/main/pkg/controllers/openshift-controller-manager.go#L52

#### What you expected to happen:
Wait for `openshift-apiserver to become ready`.

### PR DESCRIPTION
**Pull request by oglok**
Signed-off-by: Ricardo Noriega <rnoriega@redhat.com>

**Which issue(s) this PR addresses**:

Closes #515

### CODE REVIEW

**Comment by copejon** [comment]
The error in the e2e-rpm-install test is a happening because this patch's kubelet version (1.21) does not support (or have the FeatureGate enabled for) `SizeMemoryBackedVolumes`.  The latest vendored version is 1.23, where the feature is now beta and enabled by default.

**Comment by ggiguash** [comment]
USHIFT-21

File: pkg/controllers/openshift-apiserver.go [hunk 1]
```diff
@@ -45,10 +45,8 @@ func NewOpenShiftAPIServer(cfg *config.MicroshiftConfig) *OCPAPIServer {
 	return s
 }
 
-func (s *OCPAPIServer) Name() string { return "ocp-apiserver" }
-func (s *OCPAPIServer) Dependencies() []string {
-	return []string{"kube-apiserver", "openshift-crd-manager"}
-}
+func (s *OCPAPIServer) Name() string           { return "ocp-apiserver" }
```
Comment (fzdarsky) [inline pkg/controllers/openshift-apiserver.go:48]: Nitpicking here, but shall we rename this `openshift-apiserver` to correspond with the component + repo name?

Comment (copejon) [reply pkg/controllers/openshift-apiserver.go:48]: +1 improving readability

Comment (oglok) [reply pkg/controllers/openshift-apiserver.go:48]: I'll address these changes in a following PR.

**Pull request merged**